lib/developer_portal/app/controllers/developer_portal/admin/applications/alerts_controller.rb

```diff
@@ -61,15 +61,18 @@ def authorize_alerts
   end
 
   def find_application
-    @cinstance = current_account.bought_cinstances.find(params[:application_id])
+    @cinstance = current_account.bought_cinstances.find(params.require(:application_id))
```

Comment:
What kind of warning errors does this solve? I am not aware of it. The only thing I could think of is:

```ruby
params = { application_id: {table: 'column'} 
```

So basically you might prevent from some SQL injection. AFAIK `params[:application_id]` still works if the parameter is a simple value (Integer, String ...)

Reply:
> What kind of warning errors does this solve? 

None, it tries to improve the logic of the parameters. If the parameter is missing it will return an error with the message `param is missing or the value is empty`.

> So basically you might prevent from some SQL injection. 

`params.require(:application_id)` would return unpermitted hash in this context. Which would raise an error if anyone tries to convert it to a hash.

I am not sure which method is better, I like the require because it makes raises the error if it's missing